Update `flake.nix` to use `inputs` rather than `fetchFromGithub`.

sbrow opened the pull request:
This handles dependencies in the more idiomatic "flake" way. The particular revision you're referencing will be stored in the `flake.lock` file, but if you need to reference a particular branch or revision, you can do so explictly as well:

```nix
# Get the latest revision from the main branch as of the last call to `nix flake update`
inputs.dolt.url = "github:dolthub/dolt-nix-flake";

# Get the latest revision from the "taylor/create-dolt-docs" branch as of the last call to `nix flake update`
inputs.dolt.url = "github:dolthub/dolt-nix-flake?ref=taylor/create-dolt-docs";


# Get revision "https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html#examples" 
inputs.dolt.url = "github:dolthub/dolt-nix-flake/41237a70c9c065f8220322bde383ba43135433b7";
```

You can see more examples [In the docs](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html#examples).

sbrow commented:
Unfortunately, this build doesn't currently run, I'm guessing you'd have to copy the source into a new directory before running the build, which feels like taking three lefts instead of a right, so I'll leave it up to you whether on not I should put more time into this PR :)

reltuk commented:
I played with it a bit and was able to make flake inputs for the source work if I set `src = dolt` and `modRoot = './go'`. Take a look at the current contents of main and let me know what you think :).

sbrow committed:
WIP: Attempt to make package runnable with `nix run`.

sbrow commented:
It's a great start! The only issue I currently see is that the instructions assume you have knowledge of go, and have go installed in your path. Since the package is for nix users of dolt, who may not even know that dolt is built with go, I think it should be updated using `nix run .#dolt-nix-flake` rather than `go run .`.

I've started working on this, but the  `main.go` script complains that `go executable is not found in $PATH`.

reltuk committed:
flake.nix.template: Add a devShell which has the dependencies of the updater program.

Also updates the README to refer to `nix develop` instead of relying on a Go
toolchain and various other programs being available in the user's default
PATH.

reltuk commented:
That's a really good point. I'm not exactly sure of how to best model this in `nix`, but I took a pass at something I thought might be an improvement:

https://github.com/sbrow/dolt-nix-flake/pull/1

Basically, we put in a dev shell derivation that has the program derivation you just added as an input, and it has all the binaries that `main.go` is actually going to run and stuff as inputs. Testing with `nix develop -i ...` does seem to work. I also updated the README to reference `nix develop` instead of `go run .`.

If you think all that's reasonable, feel free to merge my PR into your branch and then I'll merge this PR, which is awesome.

I've been pretty busy with other stuff recently, but with the update of this nix.flake file just one hermetic `nix develop` invocation away, it will hopefully be even easier to write the GitHub actions to keep this flake up to date when I get around to it :).

sbrow committed:
Merge pull request #1 from dolthub/sbrow-fix

flake.nix.template: Add a devShell which has the dependencies of the updater program.
